Add a d3d11 wrapper

msiglreith opened the pull request:
Bindings to d3d11 (including the Video API) and the structs/enums/constants.
I tested it with a small demo application with a msvc rust build. (MinGW needs additional libraries?)


NOTE: Requires changes to the interface macro `RIDL!` and https://github.com/retep998/winapi-rs/pull/164 for building.

* Temporary workaround for an univariant enum(https://github.com/rust-lang/rust/issues/10292)
https://github.com/msiglreith/winapi-rs/blob/8210ef6df600abe9f77d65aa9c2b9e5697dbe383/src/d3d11.rs#L1263

* Some of the flags are wrapped in enums. Should I change them to `pub const` for easier use?

retep998 commented on lib/d3d11-sys/Cargo.toml:
```diff
@@ -1,15 +1,20 @@
 [package]
 name = "d3d11-sys"
-version = "0.0.1"
+version = "0.1.0"
```
Don't change the crate version. I do that when I publish things to crates.io

msiglreith commented:
Thanks! Reverted the version change and rebased onto master. Building should be successful.

retep998 commented on lib/d3d11-sys/src/lib.rs:
```diff
@@ -10,8 +10,19 @@ extern "system" {
-    // pub fn D3D11CreateDevice();
-    // pub fn D3D11CreateDeviceAndSwapChain();
+
+    pub fn D3D11CreateDevice(pAdapter: *const IDXGIAdapter, DriverType: D3D_DRIVER_TYPE,
```
Please use the multiline function style as dictated in `CONTRIBUTING.md`.

retep998 commented:
Please use `#[repr(i32)]` for the enums.

retep998 commented on lib/d3d11-sys/Cargo.toml:
```diff
@@ -2,14 +2,19 @@
-winapi = "*"
+winapi = { version = "*", path = "../.." }
+[build-dependencies]
+winapi-build = { version = "*", path = "../../build" }
```
Missing end of file newline.

msiglreith commented:
Thanks for reviewing.
* Fixed EOF newline
* Used the new `ENUM!` and `FLAGS!` macros (at some places I had to use `.0`)
* Added some enum fields and structs from the latest Win10 SDK (Direct3D 11.3)
